init.py

```diff
@@ -0,0 +1,577 @@
+    # dynamically passed in. We could parse the config to find out what is
+    # defined, but we need to interpret what the container was launched to
+    # connect to, which is likely what was passed in via environment vars.
+    possible_backends = []
+
+    # Here I _could_ just say possible_backends = ['ldap', 'ad', 'demo'], but
+    # I want to make this as low-maintenance as possible. If there is another
+    # backend that becomes available, this will catch it as-is.
+    for option in all_ini_settings['backends']:
+        possible_backend = option.split('.')[0]
+        if (possible_backend not in possible_backends and
+                possible_backend != 'demo'):
+            possible_backends.append(possible_backend)
+
+    # Get all the backends that were passed in with environment vars.
+    used_backends = []
+    for backend in possible_backends:
+        backend_prefix = "{}_".format(backend.upper())
+        if any(env_var.startswith(backend_prefix) for env_var in os.environ):
+            used_backends.append(backend)
+
+    return possible_backends, used_backends
+
+
+def include_exclude_ini_backends(config, possible_backends, used_backends,
+        all_ini_settings):
+    """
+    Allow sane defaults to be applied to the intended backend, and remove
+    unnecessary defaults from the config to avoid backend errors.
+
+    @dict config: the configuration
+    @list possible_backends: the backends that can be used
+    @list used_backends: the backends that are being used
+    @dict all_ini_settings: all of the settings possible in the ini file
+    """
+    # Insert all of the demo settings if there were no backends specified
+    if len(used_backends) == 0:
+        # Add all of the demo settings in there.
+        for option in all_ini_settings['backends']:
+            if option.startswith('demo') and option not in config['backends']:
+                config['backends'][option] = all_ini_settings['backends'][option]
+
+        # Set the admin user's security group to allow for all admin actions
+        config['backends']['demo.admin.groups'] = "'SECOFF'"
+
+        # Remove all other options for all other backends
+        for backend in possible_backends:
+            if backend != 'demo':
+                config = exclude_ini_backend(backend, config)
+
+    else:
+        # Take out all of the backend options that are not in use
+        unused_backends = list(set(possible_backends) - set(used_backends))
+        for backend in unused_backends:
+            config = exclude_ini_backend(backend, config)
+
+    return config
+
+
+def get_config_value(env_var):
+    """
+    Format valid configuration values based on the string that is passed
+
+    @str env_var: the environment var that needs to be formatted
+    """
+    val = os.getenv(env_var)
+
+    if val is not None:
+        if val.isdigit():
+            return val
+        if val in ['True', 'False']:
+            return val
```

Comment:
Shouldn't this be `bool(val)`

Reply:
Atleast for yaml files `True` will result in string True inside the file instead of boolean True.